frontend/src/Frontend/UI/Dialogs/Send.hs

```diff
@@ -252,25 +274,40 @@ sameChainTransfer model netInfo keys (fromName, fromChain, fromAcc) (gasPayer, g
 -- | General transfer workflow. This is the initial configuration screen.
 sendConfig
   :: SendConstraints model mConf key t m
-  => model -> (AccountName, ChainId, Account) -> Workflow t m (mConf, Event t ())
-sendConfig model fromAccount@(fromName, fromChain, fromAcc) = Workflow $ do
+  => model -> InitialTransferData -> Workflow t m (mConf, Event t ())
+sendConfig model initData = Workflow $ do
   close <- modalHeader $ text "Send"
   rec
-    (currentTab, _done) <- makeTabs $ attachWithMaybe (const . void . hush) (current recipient) next
+    (currentTab, _done) <- makeTabs initData $ attachWithMaybe (const . void . hush) (current recipient) next
     (conf, mCaps, recipient) <- mainSection currentTab
     (cancel, next) <- footerSection currentTab recipient mCaps
   let nextScreen = flip push next $ \() -> runMaybeT $ do
-        (fromGasPayer, mToGasPayer) <- MaybeT $ sample $ current mCaps
-        (toAccount, amount) <- MaybeT $ fmap hush $ sample $ current recipient
-        let mCCD = case mToGasPayer of
-              Just toGasPayer | toGasPayer /= fromGasPayer -> Just $ CrossChainData
-                { _crossChainData_recipientChainGasPayer = toGasPayer
-                }
-              _ -> Nothing
-        pure $ previewTransfer model fromAccount fromGasPayer toAccount mCCD amount
+        currTab <- lift $ sample $ current currentTab
+        if (currTab /= SendModalTab_Sign)
```

Comment:
Since we end with 
```haskell
pure $ previewTransfer model transfer
```
how about having `footerSection` return `Event t SendModalTab` instead of `Event t ()` and make this be
```haskell
let nextScreen = flip push next $ \nextTab -> runMaybeT $ do
  if (nextTab /= SendModalTab_Preview)
```

That way the decision of what is the next screen is in only one place.
Right now we do two separate checks: one for the 'next' button and another for the modal body.

Reply:
Done (well, preview isn't a tab, so it became `Event t (Maybe SendModalTab)` but that's nbd. Also, conflicts resolved.